Changes to Reference

royendo opened the pull request:
New order of Reference (in prep to move Connectors to Docs)
  - Project FIles
  - OLAP Engines
  - Connectors
  - CLI Usage
Rill Extensions

- Reorderd the Project Files with Connectors First (first item in the experience). 
- Added a deprecated note in source and have them look at model YAML instead
- Model YAML has a snippet on why YAML vs SQL 

Didn't add:
- MotherDuck as an OLAP engine, but it makes sense to and will do it in the future

Notes on PR:
Took the chance to recreate our **connector YAML** in preparation for the new connector experience and the importance to have this clearly defined and allow user to easily copy and paste the sample connector (or use the UI). There's a lot of parameters that we dont have documented anywhere and the **reference** should be the place for it. (not docs or relying on them to read github code)


Driver Types are defined with a in-page link to the corresponding sample YAML. This way a user can just copy and paste then insert their own creds. There is a note on each that references that they should not plain text their creds here and instead use a .env.......

ericpgreen2 reviewed (changes requested):
On the Connectors YAML page:
- I notice we don't have a right-hand side table-of-contents on this page. It'd be useful to quickly scan for the various connector types.
- It feels like the hierarchy isn't totally clear. Here's one example– there are two equivalent headers stacked on top of one another:
    <img width="230" height="68" alt="image" src="https://github.com/user-
attachments/assets/70b5a684-0460-48a3-b788-c2ed6d6e3614" />
    It might be worth asking Cursor to clean up the section hierarchies and make the sections "MECE" :).
- The "Never store plain text credentials in your YAML files" advice is great, but it is communicated a few too many times :). What if we either showed it at the top-level, or only showed it per-connector (which I think I favor)?

royendo commented:
> I notice we don't have a right-hand side table-of-contents on this page. It'd be useful to quickly scan for the various connector types.

Added. 

> It feels like the hierarchy isn't totally clear. Here's one example– there are two equivalent headers stacked on top of one another:

I agree, only so many headers in markdown to be linkable... I added a fourth (#####) and maybe add italics. Removed a section as it was referring to our 'hidden' docs. This makes it a bit easier on the eyes. 


> It might be worth asking Cursor to clean up the section hierarchies and make the sections "MECE" :).
> The "Never store plain text credentials in your YAML files" advice is great, but it is communicated a few too many times :). What if we either showed it at the top-level, or only showed it per-connector (which I think I favor)?

def overkill haha but ill add it under data source parameters and OLAP engine parameters headers

royendo commented on docs/docs/reference/project-files/connectors.md:
```diff
@@ -1,68 +1,337 @@
+- **[Google Sheets](#google-sheets)** - Public Google Sheets
 
-**`type`** - refers to the resource type and must be 'connector'
+### OLAP Engine Drivers
```
most important being an arguement for another day but done! 🤣

royendo commented on docs/docs/reference/project-files/connectors.md:
````diff
@@ -1,68 +1,337 @@
+dsn: "./data/database.db"                        # SQLite connection DSN _(required)_
+```
+
+:::warning Security Recommendation
+For all credential parameters (passwords, tokens, keys), use environment variables with the syntax `{{.env.<connector_type>.<parameter_name>}}`. This keeps sensitive data out of your YAML files and version control. See our [credentials documentation](/build/credentials/) for complete setup instructions.
+:::
````
no hard feelings for me so ill remove it for now ( can always re add any time)

ericpgreen2 reviewed (changes requested):
Okay, I think these are my last change requests!

1. It looks a bit off that these _connector types_ are listed underneath the "Required properties" heading. I think they should be their own section with a header "Connector types" (or something like that).
    <img width="168" height="176" alt="image" src="https://github.com/user-attachments/assets/ae5f81ae-5fc5-4623-858a-4f7c64ccf0e6" />
2. Are you positive that we're supposed to define the MotherDuck connection differently in these two cases?
    <img width="1992" height="928" alt="image" src="https://github.com/user-attachments/assets/e71d57e3-2455-4864-94da-1f93a6bc9d55" />

royendo commented:
1. done.
2. Yeah we have two different ways of working with MD (might need to sync with Platform to see if we deprecate legacy, but for now we support both.

Legacy is ingesting MD data into Rill via the #as-a-source and this will result in data in our Rill Embed DuckDB 
vs. 
Live connector where we'd use the user's MD (like CH and Druid) and run all of the compute on the user's side.
